presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -984,9 +996,19 @@ private void uploadObject()
 
                 PutObjectRequest request = new PutObjectRequest(host, key, tempFile);
                 if (sseEnabled) {
-                    ObjectMetadata metadata = new ObjectMetadata();
-                    metadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
-                    request.setMetadata(metadata);
+                    if (!sseKmsEnabled) {
+                        ObjectMetadata metadata = new ObjectMetadata();
+                        metadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
+                        request.setMetadata(metadata);
+                    }
+                    else {
+                        if (sseKmsKeyId.toLowerCase().equals(DEFAULT_SSE_KMS_KEY_ID.toLowerCase())) {
```

Comment:
I think this will throw `NullPointerException` if `sseKmsKeyId` is not set, since the default is null. We can remove `DEFAULT_SSE_KMS_KEY_ID` and check for null here. This also fits better with how we do config, as having a magic string like "DEFAULT" is something we avoid.